### PR TITLE
Upgrade travis platforms

### PR DESCRIPTION
**Pull request by vadz**
As discussed in [this thread](https://groups.google.com/forum/#!topic/wx-dev/t6Cwco0D6Ho)

### CODE REVIEW

File: build/tools/travis-ci.sh [hunk 1]
```diff
@@ -54,7 +54,14 @@ case $wxTOOLSET in
         ;;
     *)
         echo 'Configuring...' && echo -en 'travis_fold:start:script.configure\\r'
-        ./configure --disable-optimise $wxCONFIGURE_FLAGS
+        ./configure --disable-optimise --disable-debug_info $wxCONFIGURE_FLAGS || rc=$?
+        if [ $rc != 0 ]; then
```
Comment (paulcor) [inline build/tools/travis-ci.sh:58]: `build/tools/travis-ci.sh: line 58: [: !=: unary operator expected`

Comment (vadz) [reply build/tools/travis-ci.sh:58]: Sorry, could you please explain where do you see this? I must be blind, but I don't see anything wrong with this test...

Comment (paulcor) [reply build/tools/travis-ci.sh:58]: line 1305: [https://travis-ci.org/github/wxWidgets/wxWidgets/jobs/678182859](url)

`rc` is not set if configure return code is 0.

Comment (vadz) [reply build/tools/travis-ci.sh]: Thanks, I've committed an ugly fix in 3fc119f564 (there must be a better way to do it, but somehow I don't see it at the moment...).

File: .travis.yml [hunk 2]
```diff
@@ -41,22 +41,26 @@ matrix:
           compiler: clang
           env: wxTOOLSET=cmake wxCMAKE_GENERATOR=Xcode wxCMAKE_DEFINES="-DCMAKE_CXX_STANDARD=11"
           name: wxOSX Xcode 9.4
```
Comment (MaartenBent) [inline .travis.yml:43]:
```suggestion
          name: wxOSX CMake Xcode 9.4
```

File: build/tools/travis-ci.sh [hunk 3]
```diff
@@ -54,7 +54,14 @@ case $wxTOOLSET in
         ;;
     *)
         echo 'Configuring...' && echo -en 'travis_fold:start:script.configure\\r'
-        ./configure --disable-optimise $wxCONFIGURE_FLAGS
+        ./configure --disable-optimise --disable-debug_info $wxCONFIGURE_FLAGS || rc=$?
+        if [ "$rc" != 0 ]; then
```
Comment (paulcor) [inline build/tools/travis-ci.sh:58]: Still not right. How about this:
```
./configure --disable-optimise --disable-debug_info $wxCONFIGURE_FLAGS
rc=$?
if [ $rc != 0 ]; then
```
```

Comment (paulcor) [reply build/tools/travis-ci.sh:58]: Ahh, no, there is a `set -e`. Maybe this:
```
if [ -n "$rc" ]; then
```

Comment (vadz) [reply build/tools/travis-ci.sh]: Yes, this is what I ended up by doing, sorry for the flurry of commits.

I still have no idea why does GTK 3 STL build fail though.

File: .travis.yml (hunk 2 above)
Comment (vadz) [reply .travis.yml]: Thanks, done.

**Comment by MaartenBent** [comment]
I can reproduce the GTK3 STL problem locally. 

Should `libwx_baseu_xml` be in `objdump -p test_gui`?
```
Dynamic Section:
  NEEDED               libwx_gtk3u_stc-3.1.so.4
  NEEDED               libwx_gtk3u_aui-3.1.so.4
  NEEDED               libwx_gtk3u_richtext-3.1.so.4
  NEEDED               libglib-2.0.so.0
  NEEDED               libwx_gtk3u_xrc-3.1.so.4
  NEEDED               libwx_gtk3u_html-3.1.so.4
  NEEDED               libwx_gtk3u_core-3.1.so.4
  NEEDED               libwx_baseu_net-3.1.so.4
  NEEDED               libwx_baseu-3.1.so.4
  NEEDED               libm.so.6
  NEEDED               libstdc++.so.6
  NEEDED               libgcc_s.so.1
  NEEDED               libpthread.so.0
  NEEDED               libc.so.6
  RUNPATH              /home/maarten/build/wxWidgets/cfg/lib
...
```

`ldd test_gui`:
```
	linux-vdso.so.1 (0x00007ffd5c7be000)
	libwx_gtk3u_stc-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_gtk3u_stc-3.1.so.4 (0x00007f15da368000)
	libwx_gtk3u_aui-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_gtk3u_aui-3.1.so.4 (0x00007f15da078000)
	libwx_gtk3u_richtext-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_gtk3u_richtext-3.1.so.4 (0x00007f15d9c18000)
	libglib-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f15d9901000)
	libwx_gtk3u_xrc-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_gtk3u_xrc-3.1.so.4 (0x00007f15d95a6000)
	libwx_gtk3u_html-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_gtk3u_html-3.1.so.4 (0x00007f15d9268000)
	libwx_gtk3u_core-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_gtk3u_core-3.1.so.4 (0x00007f15d8688000)
	libwx_baseu_net-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_baseu_net-3.1.so.4 (0x00007f15d8428000)
	libwx_baseu-3.1.so.4 => /home/maarten/build/wxWidgets/cfg/lib/libwx_baseu-3.1.so.4 (0x00007f15d7ee9000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f15d7b4b000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f15d77c2000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f15d75aa000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f15d738b000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f15d6f9a000)
	libgdk-3.so.0 => /usr/lib/x86_64-linux-gnu/libgdk-3.so.0 (0x00007f15d6ca4000)
	libgtk-3.so.0 => /usr/lib/x86_64-linux-gnu/libgtk-3.so.0 (0x00007f15d639c000)
	libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f15d6148000)
	libwx_baseu_xml-3.1.so.4 => not found
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f15d5ed6000)
	libwx_baseu_xml-3.1.so.4 => not found
	libpangocairo-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0 (0x00007f15d5cc9000)
...
```

**Comment by vadz** [comment]
> I can reproduce the GTK3 STL problem locally.

I can't, but I don't have Bionic VM yet...

> Should `libwx_baseu_xml` be in `objdump -p test_gui`?

It should be in `objdump -p` output, of course, the question is why isn't it? The linker command line includes it here. But then I do see it in `objdump` output too (FWIW I prefer the output from `readelf -Wd` but they show the same thing anyhow).

Anyhow, it looks like `libwx_baseu_xml` is not linked into the test, but as it's required by one of the libraries that it links with, it is then not found during run-time. But I just don't understand why. Could you please check if it's present in the link command line for you? TIA!

**Comment by MaartenBent** [comment]
Yes, it's there. It happens also with just `configure` or `configure --with-gtk=2`. And also with 3.1.3.
```
g++ -std=gnu++11 -o test_gui  test_gui_asserthelper.o test_gui_test.o test_gui_testableframe.o test_gui_rect.o test_gui_size.o test_gui_point.o test_gui_region.o test_gui_bitmap.o test_gui_colour.o test_gui_ellipsization.o test_gui_measuring.o test_gui_affinematrix.o test_gui_boundingbox.o test_gui_clippingbox.o test_gui_graphmatrix.o test_gui_graphpath.o test_gui_config.o test_gui_auitest.o test_gui_bitmapcomboboxtest.o test_gui_bitmaptogglebuttontest.o test_gui_bookctrlbasetest.o test_gui_buttontest.o test_gui_checkboxtest.o test_gui_checklistboxtest.o test_gui_choicebooktest.o test_gui_choicetest.o test_gui_comboboxtest.o test_gui_dataviewctrltest.o test_gui_datepickerctrltest.o test_gui_frametest.o test_gui_gaugetest.o test_gui_gridtest.o test_gui_headerctrltest.o test_gui_htmllboxtest.o test_gui_hyperlinkctrltest.o test_gui_itemcontainertest.o test_gui_label.o test_gui_listbasetest.o test_gui_listbooktest.o test_gui_listboxtest.o test_gui_listctrltest.o test_gui_listviewtest.o test_gui_markuptest.o test_gui_notebooktest.o test_gui_ownerdrawncomboboxtest.o test_gui_pickerbasetest.o test_gui_pickertest.o test_gui_radioboxtest.o test_gui_radiobuttontest.o test_gui_rearrangelisttest.o test_gui_richtextctrltest.o test_gui_searchctrltest.o test_gui_simplebooktest.o test_gui_slidertest.o test_gui_spinctrldbltest.o test_gui_spinctrltest.o test_gui_styledtextctrltest.o test_gui_textctrltest.o test_gui_textentrytest.o test_gui_togglebuttontest.o test_gui_toolbooktest.o test_gui_treebooktest.o test_gui_treectrltest.o test_gui_treelistctrltest.o test_gui_virtlistctrltest.o test_gui_webtest.o test_gui_windowtest.o test_gui_dialogtest.o test_gui_clone.o test_gui_evtlooptest.o test_gui_propagation.o test_gui_keyboard.o test_gui_exec.o test_gui_fonttest.o test_gui_image.o test_gui_rawbmp.o test_gui_htmlparser.o test_gui_htmlwindow.o test_gui_htmprint.o test_gui_accelentry.o test_gui_menu.o test_gui_guifuncs.o test_gui_selstoretest.o test_gui_garbage.o test_gui_safearrayconverttest.o test_gui_settings.o test_gui_socket.o test_gui_tlw.o test_gui_dataview.o test_gui_rowheightcachetest.o test_gui_boxsizer.o test_gui_gridsizer.o test_gui_wrapsizer.o test_gui_toplevel.o test_gui_valnum.o test_gui_valtext.o test_gui_clientsize.o test_gui_setsize.o test_gui_xrctest.o     -L/home/maarten/build/wxWidgets/cfgtr/lib -Wl,-rpath,/home/maarten/build/wxWidgets/cfgtr/lib    -pthread   -lwx_gtk3u_stc-3.1 -lwx_gtk3u_aui-3.1  -lwx_gtk3u_richtext-3.1  -lwx_gtk3u_media-3.1 -lgstvideo-1.0 -lgstbase-1.0 -lgstreamer-1.0 -lgobject-2.0 -lglib-2.0 -lwx_gtk3u_xrc-3.1  -lwx_baseu_xml-3.1 -lexpat -lwx_gtk3u_html-3.1  -lwx_gtk3u_core-3.1  -lwx_baseu_net-3.1  -lwx_baseu-3.1        -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lgthread-2.0 -pthread -lglib-2.0 -lX11 -lXxf86vm -lSM -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lnotify -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lXtst -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype -lpng -lz -ljpeg -ltiff -llzma -ljbig  -lwxregexu-3.1  -pthread   -Wl,--version-script,/home/maarten/build/wxWidgets/cfgtr/version-script -lz -ldl -lsecret-1 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -llzma -lm  -lz -ldl -lsecret-1 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -llzma -lm

```

**Comment by vadz** [comment]
Thanks, but unfortunately this doesn't help and I still just don't see what could explain this... I'll have to install Bionic here to try to debug what's going on.

Note that it does work in GTK 2 build on Travis.

**Comment by MaartenBent** [comment]
> Note that it does work in GTK 2 build on Travis.

Those are monolithic and static.

**Comment by MaartenBent** [comment]
It also happens to xrcdemo
`
/xrcdemo: error while loading shared libraries: libwx_gtk3u_html-3.1.so.4: cannot open shared object file: No such file or directory
`

**Comment by vadz** [comment]
OK, using `-Wl,--disable-new-dtags` provides a workaround. But what's the new way of doing this correctly?

**Comment by vadz** [comment]
IIUC the problem is that ld 2.30, used under Bionic, doesn't consider executable rpath when searching for the indirect dependencies, i.e. dependencies of the shared libraries used by the executable. This makes sense and probably means that we need to build our libraries themselves with the correct rpath, so that they find their dependencies in the same directory.

Except it doesn't explain why this linker doesn't record the dependency in the executable. Especially when ld 2.31 does do it.

**Comment by vadz** [comment]
And the answer to the last question is that Bionic gcc is built with `--as-needed` in its `link` spec for whatever reason. so using `-Wl,--no-as-needed` works too. I'll just do this for now, to be able to (hopefully) finally merge this PR, but a better solution will be to link our shared libraries with correct rpath, of course.

**Pull request merged**